src/pydvl/valuation/samplers/classwise.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import annotations
+
+from typing import Callable, Generator
+
+import numpy as np
+from numpy.typing import NDArray
+
+from pydvl.valuation.dataset import Dataset
+from pydvl.valuation.samplers.base import EvaluationStrategy, IndexSampler
+from pydvl.valuation.samplers.powerset import PowersetSampler
+from pydvl.valuation.samplers.utils import StochasticSamplerMixin
+from pydvl.valuation.types import CSSample, IndexSetT, SampleGenerator
+from pydvl.valuation.utility.base import UtilityBase
+
+__all__ = ["ClasswiseSampler", "get_unique_labels"]
+
+
+def get_unique_labels(array: NDArray) -> NDArray:
+    """Labels of the dataset."""
+    # Object, String, Unicode, Unsigned integer, Signed integer, boolean
+    if array.dtype.kind in "OSUiub":
+        return np.unique(array)
+    raise ValueError("Dataset must be categorical to have unique labels.")
+
+
+class ClasswiseSampler(StochasticSamplerMixin, IndexSampler):
+    def __init__(
+        self,
+        in_class: IndexSampler,
+        out_of_class: PowersetSampler,
+        *,
+        min_elements_per_label: int = 1,
+        max_repeated_ooc_sampling: int = 100,
+    ):
+        super().__init__()
+        self.in_class = in_class
+        self.out_of_class = out_of_class
+        self.min_elements_per_label = min_elements_per_label
+        self.max_repeated_ooc_sampling = max_repeated_ooc_sampling
+
+    def interrupt(self):
+        self.in_class.interrupt()
+        self.out_of_class.interrupt()
+
+    def from_data(self, data: Dataset) -> Generator[list[CSSample], None, None]:
```

Comment:
When calling and reading `sampler.from_data`, I would expect to get a new or modified sampler object.

Reply:
I think we discussed this already and decided for now to leave it as is.